uint32 index
uint32 size
---
uint8[] real_data
